src/NetTopologySuite/Algorithm/RobustLineIntersector.cs

```diff
@@ -104,112 +126,107 @@ public override int ComputeIntersect(Coordinate p1, Coordinate p2, Coordinate q1
             {
-                IntersectionPoint[0] = CopyWithZInterpolate(q1, p1, p2);
-                IntersectionPoint[1] = CopyWithZInterpolate(q2, p1, p2);
+                IntersectionPoint[0] = em.CopyWithZ(q1);
+                IntersectionPoint[1] = em.CopyWithZ(q2);
                 return CollinearIntersection;
             }
             if (p1inQ && p2inQ)
             {
-                IntersectionPoint[0] = CopyWithZInterpolate(p1, q1, q2);
-                IntersectionPoint[1] = CopyWithZInterpolate(p2, q1, q2);
+                IntersectionPoint[0] = em.CopyWithZ(p1);
+                IntersectionPoint[1] = em.CopyWithZ(p2);
                 return CollinearIntersection;
             }
             if (q1inP && p1inQ)
             {
                 // if pts are equal Z is chosen arbitrarily
-                IntersectionPoint[0] = CopyWithZInterpolate(q1, p1, p2);
-                IntersectionPoint[1] = CopyWithZInterpolate(p1, q1, q2);
+                IntersectionPoint[0] = em.CopyWithZ(q1);
+                IntersectionPoint[1] = em.CopyWithZ(p1);
                 return q1.Equals(p1) && !q2inP && !p2inQ ? PointIntersection : CollinearIntersection;
             }
             if (q1inP && p2inQ)
             {
                 // if pts are equal Z is chosen arbitrarily
-                IntersectionPoint[0] = CopyWithZInterpolate(q1, p1, p2);
-                IntersectionPoint[1] = CopyWithZInterpolate(p2, q1, q2);
+                IntersectionPoint[0] = em.CopyWithZ(q1);
+                IntersectionPoint[1] = em.CopyWithZ(p2);
                 return q1.Equals(p2) && !q2inP && !p1inQ ? PointIntersection : CollinearIntersection;
             }
             if (q2inP && p1inQ)
             {
                 // if pts are equal Z is chosen arbitrarily
-                IntersectionPoint[0] = CopyWithZInterpolate(q2, p1, p2);
-                IntersectionPoint[1] = CopyWithZInterpolate(p1, q1, q2);
+                IntersectionPoint[0] = em.CopyWithZ(q2);
+                IntersectionPoint[1] = em.CopyWithZ(p1);
                 return q2.Equals(p1) && !q1inP && !p2inQ ? PointIntersection : CollinearIntersection;
             }
             if (q2inP && p2inQ)
             {
                 // if pts are equal Z is chosen arbitrarily
-                IntersectionPoint[0] = CopyWithZInterpolate(q2, p1, p2);
-                IntersectionPoint[1] = CopyWithZInterpolate(p2, q1, q2);
+                IntersectionPoint[0] = em.CopyWithZ(q2);
+                IntersectionPoint[1] = em.CopyWithZ(p2);
                 return q2.Equals(p2) && !q1inP && !p1inQ ? PointIntersection : CollinearIntersection;
             }
             return NoIntersection;
         }
 
-        private static Coordinate CopyWithZInterpolate(Coordinate p, Coordinate p1, Coordinate p2)
-        {
-            return CopyWithZ(p, zGetOrInterpolate(p, p1, p2));
-        }
 
         private static Coordinate CopyWithZ(Coordinate p, double z)
```

Comment:
this `CopyWithZ` is unused now. It's the only place* that directly uses `ExtraDimensionalCoordinate` outside of where it was originally intended to be used, so I'm in favor of deleting it

**besides tests, which I have a different comment about*

Reply:
I'd suggest to comment it out and leave a comment. Makes it easier when changes are made upstream.